### DISCUSSION TITLE
Does UnitsNet support automated scaling? (e.g. 12345 Bytes to 12.345 kB)

### DISCUSSION THREAD

**Discussion by schittli** [angularsen/UnitsNet#1166] [Q&A]
Good evening

unfortunately, I was not able to find any information about the most basic use case related to Units: Automatic scaling of units / numbers, so that it's human readable.

Example:
1. If we use `Information.FromBytes(12345)` how do we automatically get `12.345 kB`?
2. If we use `Information.FromBytes(12345000)` how do we automatically get `12.345 MB`?

So one should not have to find out if one has to call `.Megabytes`, so that the display is well readable for humans.

Thanks a lot, kind regards,
Thomas

**Comment by angularsen** [comment]
Hi, sure. See https://github.com/angularsen/UnitsNet/wiki/String-Formatting.

```cs
Information.FromBytes(12345)
    .ToUnit(InformationUnit.Kilobyte)
    .ToString("f3", CultureInfo.GetCultureInfo("en-US")); // "12.345 kB"
```

**Reply by angularsen** [reply]
As for automatic, no there is no automatically picked unit. You would have to create your own unit picker based on value and contribute it to UnitsNet if you think it would be useful for others.

The above example could be:
```cs
Information.FromBytes(12345)
    .ToNaturalUnit(InformationUnit.Kilobyte) // new method or extension method
    .ToString("f3", CultureInfo.GetCultureInfo("en-US")); // "12.345 kB"
```
